File: Answer/1161.最大层内元素和.cs
/*
 * @lc app=leetcode.cn id=1161 lang=csharp
 *
 * [1161] 最大层内元素和
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution
{
    public int MaxLevelSum(TreeNode root)
    {
        var max = 0;
        var sum = new List<int>();
        Dfs(root, 0);
        for (int i = 0; i < sum.Count; i++)
        {
            if (sum[i] > sum[max])
            {
                max = i;
            }
        }
        return max + 1;

        void Dfs(TreeNode root, int level)
        {
            if (root != null)
            {
                if (sum.Count > level)
                {
                    sum[level] += root.val;
                }
                else
                {
                    sum.Add(root.val);
                }
                Dfs(root.left, level + 1);
                Dfs(root.right, level + 1);
            }
        }
    }
}
// @lc code=end

File: Answer/1161.最大层内元素和.py
#
# @lc app=leetcode.cn id=1161 lang=python3
#
# [1161] 最大层内元素和
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from typing import List, Optional


class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        max = 0
        sum = []
        self.dfs(root, 0, sum)
        for i in range(len(sum)):
            if sum[i] > sum[max]:
                max = i
        return max + 1

    def dfs(self, root: Optional[TreeNode], level: int, sum: List[int]) -> None:
        if root is not None:
            if len(sum) > level:
                sum[level] += root.val
            else:
                sum.append(root.val)
            self.dfs(root.left, level + 1, sum)
            self.dfs(root.right, level + 1, sum)
# @lc code=end

